recirq/quantum_chess/quantum_board_test.py

```diff
@@ -1412,3 +1412,71 @@ def test_split_capture_with_failed_measurement_outcome(board):
         probs = b.get_probability_distribution(100)
         assert_prob_about(probs, u.square_to_bit("a2"), 1, atol=0.1)
         assert_prob_about(probs, u.square_to_bit("a3"), 0, atol=0.1)
+
+
+@pytest.mark.parametrize("board", ALL_CIRQ_BOARDS)
+def test_merge_to_fully_classical_position(board):
+    """Splits piece on d4 to b3 and c2, then merge back to fully classical position on a1."""
+    b = simulator(u.squares_to_bitboard(["d4"]))
+    b.reset_starting_states = True
+    assert b.perform_moves(
+        "d4^b3c2:SPLIT_JUMP:BASIC",
+        "b3c2^a1:MERGE_JUMP:BASIC",
+    )
+
+    assert b.is_classical()
```

Comment:
It would be good to check that the bitboard equals the expected one here

Reply:
I changed it so that it checks to make sure the probabilities are the same now